lib/Pool.js

```diff
@@ -148,6 +150,15 @@ class Pool extends EventEmitter {
     this._ensureMinimum();
   }
 
+  _applyDestroyTimeout(promise) {
+    const timeoutPromise = new this._Promise((resolve, reject) => {
+      setTimeout(() => {
+        reject(new Error("destroy timed out"));
+      }, this._config.destroyTimeoutMillis).unref();
+    });
+    return this._Promise.race([timeoutPromise, promise]);
```

Comment:
I am not 100% sure that Promise.race will be available on all supported promise libraries. I could check to see if the function exists, and skip otherwise. Alternatively I could implement without Promise.race, but it would be a bit messy.

Reply:
I think it's fine to expect the either the users environment or supplied promise library to support `Promise.race`. It's been in nodejs since ~ 4.9.1.
It's not breaking BC because `destroyTimeoutMillis` is a new feature which you'd have to opt-in to.